### ISSUE TITLE
NOT Support in SQL - Pentaho Integration

### ISSUE THREAD

**Issue by koszti** [Anchormen/sql4es#13]
I am trying to use it with Pentaho as described at https://www.anchormen.nl/elasticsearch-as-a-pentaho-source-through-sql4es/

The driver works well in SQLWorkbenchJ but Pentaho gives an error when trying to create an Analysis Report:
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.sql.SQLException: NOT is currently not supported, use '<>' instead

I'm using SQL4ES 0.8.2.3 , Elasticsearch 2.3.4 and Pentaho 6.1.0.1 Trial.
How could you make it work with Pentaho in the post? Is there any plan to add NOT support?

![image](https://cloud.githubusercontent.com/assets/643687/17109390/bdfef40c-528f-11e6-8ea6-8c720af23f3c.png)

**Comment by KBP-TDC** [comment]
Could you please provide a simple SQL statement that could reproduce this issue?

**Comment by koszti** [comment]
`SELECT di_domain_name FROM registration GROUP BY di_domain_name HAVING NOT((SUM(registrations) IS NULL))`

This is the simplified version that generated by Pentaho and it gives the same exception from SQLWorkbenchJ.

**Comment by corneversloot** [comment]
I checked this issue and noticed that the message itself is not accurate. The parser can parse the NOT clause but the sql4es driver currently cannot handle NOT within HAVING. The current implementation executes the HAVING part in the driver and not by Elasticsearch. I think this can be achieved using 'pipeline aggregations'. Unless someone else wants to dive into this i do not think i'll be able to resolve this issue anytime soon.

When I used the driver with Pentaho it did not generate the 'HAVING NOT' (apparently) so I did not encounter this issue.
